eng/targets/GeneratePkgDef.targets

```diff
@@ -145,7 +145,8 @@
     Populates PkgDefBindingRedirect and PkgDefCodeBase items from references.
   -->
   <Target Name="_AddPkgDefEntriesFromReferences"
-          BeforeTargets="AfterResolveReferences;GeneratePkgDef">
+          BeforeTargets="GeneratePkgDef"
+          DependsOnTargets="ResolveReferences">
```

Comment:
why is this more correct?

Reply:
It looks like if references are old, then pkddef files are also old.
Then references are updated, pkgdef files are not updated.

Reply:
I guess I don't understand why having a dependson ResolveReferences is that different from running before AfterResolveReferences

Reply:
@dibarbet good question!

DependsOnTargets means "before my target runs, please ensure this other target has ran". BeforeTargets means "right before that other target runs, make sure I run".

The problem here with the BeforeTargets being AfterResolveReferences;GeneratePkgDef is this target would run after reference resolution happens -- _if it happened_. If something else forced GeneratePkgDef to run sooner (or reference resolution was skipped during a particular msbuild invocation), then it would still run right before GeneratePkgDef, but since reference resolution hadn't happened (yet/at all), it wouldn't put everything in the pkgdef file that should be there. Then, a later run wouldn't write the pkgdef file at all, since it saw the pkgdef file was up-to-date from the previous run, even though it's corrupt.

Reply:
@rainersigwald